proposals/lip-0052.md

```diff
@@ -30,329 +31,345 @@ Therefore, providing a unified module to handle, transfer and modify NFTs is a n
+When an NFT is sent to another chain, the `attributes` property of the NFT can be modified according to specifications set on the receiving chain. For this reason, custom modules specifying an NFT collection must also implement the behavior to adopt when an NFT is returned with a modified `attributes` property. This custom behavior will compare the returned attributes with the ones stored with the escrowed NFT. If the returned NFT has an empty attributes, the native chain will restore the attributes as stored, this can be used to save on cross-chain messages size when returning non-modified NFTs to their native chains.
 
 ### Protocol Logic for Other Modules
 
 The NFT module provides the following functions to modify the NFT state. Any other modules should use those functions to modify the NFT state. The NFT state should never be modified from outside the module without using one of the provided functions as this could result in unexpected behavior and could cause an improper state transition.
 
 #### create
 
-This function is used to create a new NFT. The NFT will always be native to the chain creating it. The index of the created NFT will be the next available index, as specified by the max index corresponding to the collection.
+This function is used to create a new NFT. The NFT will always be native to the chain creating it.
 
 #### destroy
 
-This function is used to destroy NFTs. The NFT will be removed from the NFT store and cannot be retrieved. The use of this function is limited to destroying native NFTs.
-
-#### transfer
-
-This function is used to transfer ownership of NFTs within one chain.
-
-#### transferCrossChain
-
-This function is used to transfer ownership of NFTs across chains in the Lisk ecosystem.
+This function is used to destroy NFTs. The NFT will be removed from the NFT substore and cannot be retrieved. The use of this function is limited to destroying native NFTs.
 
 #### lock
 
-This function is used to lock an NFT to a module ID. A locked NFT cannot be transferred (within the chain or across chains). This can be useful, for example, when the NFT is used as a deposit for a service. A module ID is specified when locking the NFT and this ID has to be specified when unlocking the NFT. This avoids NFTs being accidentally locked and unlocked by different modules.
+This function is used to lock an NFT to a module. A locked NFT cannot be transferred (within the chain or across chains). This can be useful, for example, when the NFT is used as a deposit for a service. Module is specified both when locking and unlocking the NFT, thus preventing NFTs being accidentally locked and unlocked by different modules.
 
 #### unlock
 
-This function is used to unlock an NFT that was locked to a module ID.
+This function is used to unlock an NFT that was locked to a module.
 
 #### setAttributes
```

Comment:
I left a comment about this later on. I think it would be nice to explain how this method could be used to process NFTs coming back to the native chain in a ccm with possibly updated attributes. The idea then would be that the NFT module does not update the attributes, but rather the custom module should call this method from the `afterCrossChainMessageExecute` hook.

Reply:
Again, cross-chain modification of attributes is one of the topics that I would like to further discuss live.

Reply:
can be resolved